tests/sentry/relay/config/test_metric_extraction.py

```diff
@@ -1328,34 +1328,23 @@ def test_widgets_with_without_environment_do_not_collide(default_project: Projec
 
 
 @django_db_all
-def test_alert_and_widget_colliding(
-    default_project: Project, default_environment: Environment
-) -> None:
+def test_alert_and_widget_colliding(default_project: Project) -> None:
     aggr = "count()"
     query = "transaction.duration:>=10"
     condition = {"name": "event.duration", "op": "gte", "value": 10.0}
     env_tag = {"field": "event.environment", "key": "environment"}
 
     with Feature([ON_DEMAND_METRICS, ON_DEMAND_METRICS_WIDGETS]):
-        alert = create_alert(aggr, query, default_project, environment=default_environment)
         widget = create_widget([aggr], query, default_project)
 
         config = get_metric_extraction_config(default_project)
 
-        assert config and config["metrics"] == [
-            # The alert does not include the environment as part of the tags
-            _metric_spec(
-                query_hash="0af3eea3",
-                condition={
-                    "inner": [
-                        {"name": "event.environment", "op": "eq", "value": "development"},
-                        condition,
-                    ],
-                    "op": "and",
-                },
-            ),
-            _metric_spec("f1353b0f", [env_tag], condition=condition),
-        ]
+        assert config and config["metrics"] == [_metric_spec("f1353b0f", condition, [env_tag])]
```

Comment:
@k-fish This commit shows that a spec created by an alert and a widget can collide.

Line 1347 shows that the alert spec is picked, thus, `[env_tag]` is lost.